protocol/vm/context.go

```diff
@@ -0,0 +1,31 @@
+package vm
+
+// Context contains the execution context for the virtual machine.
+//
+// By convention, variables of this type have the name context, _not_
+// ctx (to avoid confusion with context.Context).
+type Context struct {
+	VMVersion uint64
+	Code      []byte
+	Arguments [][]byte
+
+	TxVersion *uint64
+
+	BlockHash            *[]byte
+	BlockTimeMS          *uint64
+	NextConsensusProgram *[]byte
+
+	TxSigHash        *[]byte
```

Comment:
Oh good point. But, I don't know, I don't much like the idea of relying on the difference between `[]byte(nil)` and `[]byte{}` because I've seen one turn into the other too easily. Maybe just pilot error, but still.